### PR TITLE
Onboarding new tests

### PR DESCRIPTION
**Pull request by burtonemily**
Adds in new tests
- Disable disappearing messages in 1:1 conversations
- Delete contact (conversation)
- Hide recovery password permanently

Removes type checking on strings returned from JSON file for locales
Adds in ENUM for Username instead of type
Starts to implement modal string checks

### CODE REVIEW

File: run/test/specs/locators/disappearing_messages.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          selector: `Disappearing messages type and time`,
+        };
+      case 'ios':
+        return {
+          strategy: 'accessibility id',
+          selector: 'Disappearing messages type and time',
+        };
+    }
+  }
+}
```
Comment (Bilb) [inline run/test/specs/locators/disappearing_messages.ts]: when the android and the ios are the same, you can just write
```ts
export class DisappearingMessagesSubtitle extends LocatorsInterface {
  public build(): StrategyExtractionObj {
    switch (this.platform) {
      case 'android':
      case 'ios':
        return {
          strategy: 'accessibility id',
          selector: `Disappearing messages type and time`,
        };
    }
  }
}
```

and both will share the same values.
(there are a few duplicates in this file, check the others too!)

File: run/test/specs/locators/index.ts [hunk 2]
```diff
@@ -499,24 +499,39 @@ export class RevealRecoveryPhraseButton extends LocatorsInterface {
...
         };
       case 'ios':
         return {
           strategy: 'accessibility id',
-          selector: 'Disappearing Messages',
+          selector: 'Download',
         };
     }
   }
 }
```
Comment (Bilb) [inline run/test/specs/locators/index.ts]: same comment

File: run/test/specs/locators/settings.ts [hunk 3]
```diff
@@ -10,20 +10,70 @@ export class HideRecoveryPasswordButton extends LocatorsInterface {
...
+        return {
+          strategy: 'accessibility id',
+          selector: 'User settings',
+        } as const;
+      case 'ios':
+        return {
+          strategy: 'accessibility id',
+          selector: 'User settings',
+        } as const;
+    }
```
Comment (Bilb) [inline run/test/specs/locators/settings.ts]: same comment

File: sessionReporter.ts [hunk 4]
```diff
@@ -11,6 +11,10 @@ import type {
 import chalk from 'chalk';
 import { Dictionary, groupBy, isString, mean, sortBy } from 'lodash';
 
+// import dotenv from 'dotenv';
+
+// dotenv.config();
+
```
Comment (Bilb) [inline sessionReporter.ts]: can be removed altogether?

File: run/types/sessionIt.ts [hunk 5]
```diff
@@ -97,27 +98,27 @@ export function bothPlatformsIt(
...
-    device3: DeviceWrapper;
-    userA: User;
-    userB: User;
-  }) => Promise<void>
-) {
-  const testName = `${title} ios`;
+// export function iosItWithSetup(
+//   title: string,
+//   setupData: SetupData,
+//   testToRun: (setupData: {
```
Comment (Bilb) [inline run/types/sessionIt.ts]: can be removed altogether?

File: run/types/sessionIt.ts [hunk 6]
```diff
@@ -34,11 +33,12 @@ export function androidIt(
...
-  shouldSkip: boolean = false
+  shouldSkip = false
 ) {
   const testName = `${title} ios`;
 
   if (shouldSkip) {
+    // eslint-disable-next-line @typescript-eslint/require-await
     test.skip(testName, async () => {
       console.info(`\n\n==========> Skipping "${testName}"\n\n`);
     });
```
Comment (Bilb) [inline run/types/sessionIt.ts]: you should be able to get rid of that  `// eslint-disable-next-line @typescript-eslint/require-await` line by removing the `async` here `    test.skip(testName, async () => {`

File: run/types/sessionIt.ts [hunk 7]
```diff
@@ -57,10 +57,11 @@ function mobileIt(
   platform: SupportedPlatformsType,
   title: string,
   testToRun: (platform: SupportedPlatformsType) => Promise<void>,
-  shouldSkip: boolean = false
+  shouldSkip = false
 ) {
   const testName = `${title} ${platform}`;
   if (shouldSkip) {
+    // eslint-disable-next-line @typescript-eslint/require-await
```
Comment (Bilb) [inline run/types/sessionIt.ts]: same comment

File: run/test/specs/disappear_after_send_off_1o1.spec.ts [hunk 8]
```diff
@@ -1,34 +1,37 @@
...
+iosIt('Disappear after send off 1:1', disappearAfterSendOff1o1);
+androidIt('Disappear after send off 1:1', disappearAfterSendOff1o1);
 
 async function disappearAfterSendOff1o1(platform: SupportedPlatformsType) {
   const { device1, device2, device3 } = await openAppThreeDevices(platform);
-  await linkedDevice(device1, device3, USERNAME.ALICE, platform);
+  const userA = await linkedDevice(device1, device3, USERNAME.ALICE, platform);
   const mode: DisappearModes = 'send';
-  const testMessage = `Checking disappear after ${mode} is working and can be turned off`;
+  const userB = await newUser(device2, USERNAME.BOB, platform);
```
Comment (Bilb) [inline run/test/specs/disappear_after_send_off_1o1.spec.ts]: could be named alice and bob :D

File: run/test/specs/linked_device_profile_picture_syncs.spec.ts [hunk 9]
```diff
@@ -24,35 +25,37 @@ async function avatarRestored(platform: SupportedPlatformsType) {
...
-  if (pixelColorLinked === expectedPixelHexColour) {
+  const base64A = await device2.getElementScreenshot(profilePictureLinked.ELEMENT);
+  const actualPixelColorLinked = await parseDataImage(base64A);
+  if (actualPixelColorLinked === expectedPixelHexColour) {
     console.log('Device 2: Colour is correct on linked device');
   } else {
-    console.log(`Device 2: Colour isn't ${expectedPixelHexColour}, it is: `, pixelColorLinked);
+    throw new Error(
+      `Device1: Colour isn't ${expectedPixelHexColour}, it is: ` + actualPixelColorLinked
+    );
```
Comment (Bilb) [inline run/test/specs/linked_device_profile_picture_syncs.spec.ts]:
```ts

if (actualPixelColorLinked !== expectedPixelHexColour) {
 throw new Error(
    `Device1: Colour isn't ${expectedPixelHexColour}, it is: ` + actualPixelColorLinked
  );
} 
  console.log('Device 2: Colour is correct on linked device');

  ```

File: run/test/specs/linked_device_profile_picture_syncs.spec.ts [hunk 10]
```diff
@@ -24,35 +25,37 @@ async function avatarRestored(platform: SupportedPlatformsType) {
...
   } else {
-    throw new Error("Device1: Colour isn't 04cbfe, it is: " + actualPixelColor);
+    throw new Error(`Device1: Colour isn't ${expectedPixelHexColour}, it is: ` + actualPixelColor);
   }
   console.log('Now checking avatar on linked device');
   // Check avatar on device 2
-  const el2 = await device2.waitForTextElementToBePresent({
+  await sleepFor(3000);
+  await device2.clickOnElementAll(new ExitUserProfile(device2));
+  await device2.clickOnByAccessibilityID('User settings');
```
Comment (Bilb) [inline run/test/specs/linked_device_profile_picture_syncs.spec.ts]: could be `new UserSettings(device2)` (and in a bunch of other places too). I guess you udpate each tests one after the other?

File: run/test/specs/utils/disappearing_control_messages.ts [hunk 11]
```diff
@@ -42,4 +43,16 @@ export const checkDisappearingControlMessage = async (
...
+  if (linkedDevice && platform === 'android') {
+    await linkedDevice.clickOnElementAll({
+      strategy: 'accessibility id',
+      selector: 'Conversation list item',
+      text: userB.userName,
+    });
+    await linkedDevice.disappearingControlMessage(disappearingMessagesSetYou);
+    await linkedDevice.disappearingControlMessage(disappearingMessagesSetUserB);
+  } else if (linkedDevice && platform === 'ios') {
+    console.log('Control message syncing is not supported on iOS');
```
Comment (Bilb) [inline run/test/specs/utils/disappearing_control_messages.ts]: not a reason to make the test fail?  I mean this is expected?

File: run/types/DeviceWrapper.ts [hunk 12]
```diff
@@ -642,40 +643,56 @@ export class DeviceWrapper {
...
     let element: AppiumNextElementType | null = null;
+    let locator: StrategyExtractionObj & { text?: string; maxWait?: number };
+    if (args instanceof LocatorsInterface) {
+      locator = args.build();
+    } else {
+      locator = args as StrategyExtractionObj & {
+        text?: string;
+        maxWait?: number;
+      };
+    }
```
Comment (Bilb) [inline run/types/DeviceWrapper.ts:662]: I should have a look at this

File: run/test/specs/user_actions_delete_contact.spec.ts [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+import { newContact } from './utils/create_contact';
+import { linkedDevice } from './utils/link_device';
+import { openAppMultipleDevices, SupportedPlatformsType } from './utils/open_app';
+
+bothPlatformsIt('Delete contact', deleteContact);
+
+async function deleteContact(platform: SupportedPlatformsType) {
+  const [device1, device2, device3] = await openAppMultipleDevices(platform, 3);
+  const userA = await linkedDevice(device1, device3, USERNAME.ALICE, platform);
+  const userB = await newUser(device2, USERNAME.BOB, platform);
```
Comment (Bilb) [inline run/test/specs/user_actions_delete_contact.spec.ts]: you should be able to await Promise.all those 2 calls

File: run/test/specs/user_actions_delete_contact.spec.ts [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+  await newContact(platform, device1, userA, device2, userB);
+  // Check contact has loaded on linked device
+  await device1.navigateBack(platform);
+  await device2.navigateBack(platform);
```
Comment (Bilb) [inline run/test/specs/user_actions_delete_contact.spec.ts]: you shouldn't need a platform argument on navigateBack, as it is a class method. You know the platform from the class instance

File: run/test/specs/user_actions_delete_contact.spec.ts [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+      selector: 'Conversation list item',
+      text: userB.userName,
+    }),
+  ]);
+  // Delete contact
+  await runOnlyOnIOS(platform, () => device1.swipeLeft('Conversation list item', userB.userName));
+  await runOnlyOnAndroid(platform, () => device1.longPressConversation(userB.userName));
+  await device1.clickOnElementAll({ strategy: 'accessibility id', selector: 'Delete' });
+  // await device1.checkModalStrings('conversationsDelete', 'conversationsDeleteDescription');
+  await device1.clickOnElementAll(new DeleteContactModalConfirm(device1));
```
Comment (Bilb) [inline run/test/specs/user_actions_delete_contact.spec.ts:42]: are those 2 the same? Should the one commented be removed?

File: run/types/DeviceWrapper.ts [hunk 16]
```diff
@@ -1753,6 +1772,68 @@ export class DeviceWrapper {
...
+    expectedStringHeading: TokenString<LocalizerDictionary>,
+    expectedStringDescription: TokenString<LocalizerDictionary>,
+    oldModalAndroid?: boolean
+    // args?: { [key: string]: string | number }
+  ) {
+    // Check modal heading is correct
+    function replaceBrWithCRLF(input: string): string {
+      // return input.replace(/<br\s*\/?>/gi, '\nCR LF');
+      return input.replace(/\n/gi, '');
+    }
```
Comment (Bilb) [inline run/types/DeviceWrapper.ts]: the name is misleading, maybe rename it to `removeNewLines`
also, isn't playwright supposed to do those checks for you?

File: run/test/specs/disappear_after_send_off_1o1.spec.ts [hunk 17]
```diff
@@ -45,51 +48,54 @@ async function disappearAfterSendOff1o1(platform: SupportedPlatformsType) {
...
+    device1.doesElementExist({
+      ...new DisappearingMessagesSubtitle(device1).build(),
+      maxWait: 500,
     }),
-    device2.hasElementBeenDeleted({
-      strategy: 'accessibility id',
-      selector: 'Message body',
-      text: testMessage,
+    device2.doesElementExist({
+      ...new DisappearingMessagesSubtitle(device2).build(),
```
Comment (Bilb) [inline run/test/specs/disappear_after_send_off_1o1.spec.ts:90]: you don't want to allow the types not built in doesElementExist?

File: .eslintrc.js [hunk 18]
```diff
@@ -14,7 +14,7 @@ module.exports = {
     "@typescript-eslint/no-unsafe-member-access": "off",
     "@typescript-eslint/no-explicit-any": "off",
     "@typescript-eslint/no-unsafe-call": "off",
-    "@typescript-eslint/no-unsafe-assignment": "off"
-
+    "@typescript-eslint/no-unsafe-assignment": "off",
+    "@typescript-eslint/no-inferrable-types": "off",
```
Comment (Bilb) [inline .eslintrc.js:18]: this might cause issues down the line, but why not

**Review by Bilb — CHANGES_REQUESTED** [review]

File: run/test/specs/locators/disappearing_messages.ts (hunk 1 above)
Comment (burtonemily) [reply run/test/specs/locators/disappearing_messages.ts]: Fixed

File: run/test/specs/linked_device_profile_picture_syncs.spec.ts (hunk 10 above)
Comment (burtonemily) [reply run/test/specs/linked_device_profile_picture_syncs.spec.ts]: Updated all instances

File: run/test/specs/utils/disappearing_control_messages.ts (hunk 11 above)
Comment (burtonemily) [reply run/test/specs/utils/disappearing_control_messages.ts]: Forgot to remove this once the bug was reported, will remove now

File: run/types/DeviceWrapper.ts (hunk 12 above)
Comment (burtonemily) [reply run/types/DeviceWrapper.ts:662]: As in you're not happy with this?

File: run/test/specs/user_actions_delete_contact.spec.ts (hunk 14 above)
Comment (burtonemily) [reply run/test/specs/user_actions_delete_contact.spec.ts]: Removed

File: run/test/specs/user_actions_delete_contact.spec.ts (hunk 15 above)
Comment (burtonemily) [reply run/test/specs/user_actions_delete_contact.spec.ts:42]: No the top line that is commented out is the function that checks if the modal strings are correct but in this PR we didn't have a way to add args into the modal checks, so i have just commented it out for now. Will add a TODO to uncommented once the fix i've implemented is merged. 
The second line is just clicking the button

File: run/types/DeviceWrapper.ts (hunk 16 above)
Comment (burtonemily) [reply run/types/DeviceWrapper.ts]: Playwright currently does not do these checks, so had to create my own function. Will rename.

File: run/test/specs/disappear_after_send_off_1o1.spec.ts (hunk 17 above)
Comment (burtonemily) [reply run/test/specs/disappear_after_send_off_1o1.spec.ts:90]: This was to allow a custom maxWait. The current implementation doesn't allow the new locator type with any extra variables. I tried a bunch of different ways and this was the best I could come up with

File: .eslintrc.js (hunk 18 above)
Comment (burtonemily) [reply .eslintrc.js:18]: This is something Ryan added, so I have no idea what it is..

File: run/test/specs/community_tests_image.spec.ts [hunk 19]
```diff
@@ -44,7 +44,7 @@ async function sendImageCommunityAndroid(platform: SupportedPlatformsType) {
   ]);
   // await newContact(platform, device1, userA, device2, userB);
   const replyMessage = `Replying to image from ${userA.userName} in community ${testCommunityName} + ${time}`;
-  // await Promise.all([device1.navigateBack(platform), device2.navigateBack(platform)]);
+  // await Promise.all([device1.navigateBack(), device2.navigateBack()]);
```
Comment (Bilb) [inline run/test/specs/community_tests_image.spec.ts]: to remove?

File: run/test/specs/group_disappearing_messages_link.spec.ts [hunk 20]
```diff
@@ -25,7 +25,7 @@ async function disappearingLinkMessageGroup(platform: SupportedPlatformsType) {
   ]);
   await createGroup(platform, device1, userA, device2, userB, device3, userC, testGroupName);
   await setDisappearingMessage(platform, device1, ['Group', timerType, time]);
-  // await device1.navigateBack(platform);
+  // await device1.navigateBack();
```
Comment (Bilb) [inline run/test/specs/group_disappearing_messages_link.spec.ts]: not sure why you keep all of those commented navigateBack

File: run/types/DeviceWrapper.ts [hunk 21]
```diff
@@ -1753,6 +1772,68 @@ export class DeviceWrapper {
...
+    const actualDescription = await this.getTextFromElement(elDescription);
+    // Need to format the ACTUAL description that comes back from device to match
+    const formattedDescription = removeNewLines(actualDescription);
+    if (expectedDescription === formattedDescription) {
+      console.log('Modal description is correct');
+    } else {
+      throw new Error(
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        `Modal description is incorrect. Expected description: ${expectedDescription}, Actual description: ${formattedDescription}`
```
Comment (Bilb) [inline run/types/DeviceWrapper.ts]: if statements could be swap to have the throw and then normal execution

**Review by Bilb — APPROVED** [review]

File: run/types/DeviceWrapper.ts (hunk 12 above)
Comment (Bilb) [reply run/types/DeviceWrapper.ts:662]: not really no. Will check it out once it's merged

File: run/test/specs/group_disappearing_messages_link.spec.ts (hunk 20 above)
Comment (burtonemily) [reply run/test/specs/group_disappearing_messages_link.spec.ts]: For fun

**Review by Bilb — APPROVED** [review]

**Pull request merged**